### PR TITLE
Enable ngraph tests for a ngraph engine

### PR DESCRIPTION
**Pull request by mozga-intel**
Introduces the nGraph's implementation of test executor.

### CODE REVIEW

**Review by luotao1 — COMMENTED** [review]
- How about create a separate directory `python/paddle/fluid/tests/unittests/ngraph/` to put all ngraph unittests?
- It's better to combine this PR into #14801, since we can see the correctness of `test_mul_ngraph_op.py` on CI.

**Comment by mozga-intel** [comment]
@luotao1 done.

File: python/paddle/fluid/tests/unittests/CMakeLists.txt [hunk 1]
```diff
@@ -111,3 +111,5 @@ py_test_modules(test_parallel_executor_transformer MODULES test_parallel_executo
 if(NOT APPLE)
     py_test_modules(test_image_classification_resnet MODULES test_image_classification_resnet SERIAL)
 endif()
+
+add_subdirectory(ngraph)
```
Comment (luotao1) [inline python/paddle/fluid/tests/unittests/CMakeLists.txt]:
```
if (WITH_NGRAPH)
  add_subdirectory(ngraph)
endif()
```

File: python/paddle/fluid/tests/unittests/ngraph/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    add_test(NAME ${TARGET_NAME}
+        COMMAND ${CMAKE_COMMAND} -E env FLAGS_use_ngraph=${use_ngraph} PYTHONPATH=${PADDLE_BINARY_DIR}/python ${py_test_modules_ENVS}
+        ${PYTHON_EXECUTABLE} ${PADDLE_SOURCE_DIR}/tools/test_runner.py ${py_test_modules_MODULES}
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
+    if (py_test_modules_SERIAL)
+        set_property(TEST ${TARGET_NAME} PROPERTY RUN_SERIAL 1)
+    endif()
+    set_tests_properties(${TARGET_NAME} PROPERTIES TIMEOUT 600)
+  endif()
+endfunction()
```
Comment (luotao1) [inline python/paddle/fluid/tests/unittests/ngraph/CMakeLists.txt]: line4-29 could be removed

File: python/paddle/fluid/tests/unittests/ngraph/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+foreach(TEST_OP ${TEST_OPS})
+    py_test_modules(${TEST_OP} MODULES ${TEST_OP})
```
Comment (luotao1) [inline python/paddle/fluid/tests/unittests/ngraph/CMakeLists.txt]:
```
py_test_modules(${TEST_OP} MODULES ${TEST_OP} ENVS FLAGS_use_ngraph=true)
```
like 
https://github.com/PaddlePaddle/Paddle/blob/c80c693a0f41c6f05f4945225c472d9e743424f0/python/paddle/fluid/tests/unittests/CMakeLists.txt#L89

**Comment by mozga-intel** [comment]
@luotao1. Done.

**Review by luotao1 — APPROVED** [review]
LGTM

**Pull request merged**